### PR TITLE
Fix NT-Password

### LINKED ISSUE #17: Unable to authenticate via EAP-PEAP-MSCHAPv2
When testing this, I was unable to use EAP-PEAP-MSCHAPv2.
I have checked both password was correct.

Digging into the logs, the formatting of NT-Hash might be the issue

```
(13) rest_passwords: Adding reply:REST-HTTP-Status-Code = "201"
(13) rest_passwords: Parsing attribute "control:NT-Password"
(13) rest_passwords: EXPAND <redacted>
(13) rest_passwords: --> <redacted>
(13) rest_passwords: NT-Password := 0x<redacted>
(13) rest_passwords: Parsing attribute "control:SSHA2-512-Password"
(13) rest_passwords: EXPAND <redacted>
(13) rest_passwords: --> <redacted>
(13) rest_passwords: SSHA2-512-Password := 0x<redacted>

(13) eap_mschapv2: Auth-Type MS-CHAP {
(13) mschap: WARNING: NT-Password has not been normalized by the 'pap' module (likely still in hex format). Authentication may fail
(13) mschap: WARNING: No Cleartext-Password configured. Cannot create NT-Password
(13) mschap: Creating challenge hash with username: test
(13) mschap: Client is using MS-CHAPv2
(13) mschap: ERROR: FAILED: No NT-Password. Cannot perform authentication
(13) mschap: ERROR: MS-CHAP2-Response is incorrect
(13) eap_mschapv2: [mschap] = reject
(13) eap_mschapv2: } # Auth-Type MS-CHAP = reject
(13) eap_mschapv2: MSCHAP-Error: ?E=691 R=1 C=fdb45badccc2323b97862a2f575ef9c7 V=3 M=Authentication rejected
(13) eap_mschapv2: Found new challenge from MS-CHAP-Error: err=691 retry=1 challenge=fdb45badccc2323b97862a2f575ef9c7
(13) eap_mschapv2: ERROR: MSCHAP Failure
```

### PR DESCRIPTION
**Pull request by DariusSG**
Added missing 0x required for setting NT-Password directly via hex

Closes #17

### CODE REVIEW

**Comment by amphineko** [comment]
The change generally looks good to me, let me merge this weekend when I'm available.

**Review by amphineko — APPROVED** [review]

**Comment by amphineko** [comment]
https://github.com/FreeRADIUS/freeradius-server/blob/d956f683d37ea40e7977cc5907361f3e6988a439/src/modules/rlm_pap/rlm_pap.c#L159-L164

```c
/** Hex or base64 or bin auto-discovery
 *
 * Here we try and autodiscover what encoding was used for the password/hash, and
 * convert it back to binary or plaintext.
 *
 * @note Earlier versions used a 0x prefix as a hard indicator that the string was
 *       hex encoded, and would fail if the 0x was present but the string didn't
 *       consist of hexits. The base64 char set is a superset of hex, and it was
 *       observed in the wild, that occasionally base64 encoded data really could
 *       start with 0x. That's why min_len (and decodability) are used as the
 *       only heuristics now.
 *
 * @param[in] request Current request.
 * @param[in,out] vp to normify.
 * @param[in] min_len we expect the decoded version to be.
 */
static void normify(REQUEST *request, VALUE_PAIR *vp, size_t min_len)
```

I think it's a good idea to prefix `0x` to all password hashes returned to radiusd.

Can you also accommodate this before I merge? Thank you.

File: packages/supervisor/src/rlm_rest/types/passwordAuth.ts
Comment (DariusSG) [reply packages/supervisor/src/rlm_rest/types/passwordAuth.ts]: Added prepending `0x` to `SSHA-Password` and `SSHA2-512-Password`

**Pull request merged**
